cluster.go

```diff
@@ -125,6 +132,33 @@ func (cfg *ClusterConfig) CreateSession() (*Session, error) {
 	return NewSession(*cfg)
 }
 
+// translateHostPort is a helper method that will use the given AddressTranslator
+// if defined, to translate the given host:port (addr) into a new host:port string.
+// If no AddressTranslator or if an error occurs, the given host:port will be returned.
+func (cfg *ClusterConfig) translateHostPort(hostPort string) (string) {
+	if cfg.AddressTranslator == nil {
+		return hostPort
+	}
+	addr, port, err := parseIPPort(hostPort)
+	if err != nil {
+		return hostPort
+	}
+	newAddr, newPort := cfg.AddressTranslator.Translate(addr, port)
+	if gocqlDebug {
+		log.Printf("gocql: translating address '%s' to '%s:%d'", hostPort, newAddr.String(), newPort)
+	}
+	return net.JoinHostPort(newAddr.String(), strconv.Itoa(newPort))
+}
+
+func parseIPPort(ipPort string) (net.IP, int, error) {
+	addr, portStr, err := net.SplitHostPort(ipPort)
+	if err != nil {
+		return nil, 0, err
+	}
+	port, _ := strconv.Atoi(portStr)
+	return net.ParseIP(addr), port, nil
```

Comment:
In the event that `addr` is a hostname, [ParseIP](https://golang.org/pkg/net/#ParseIP) will return `nil`. This in turn causes the translator to fail as it recieves `nil` for the IP.

Currently `NewCluster` supports accepting a list of host names. This change ungracefully breaks that functionality.

It may be that `NewCluster` is only intended to accept a list of IP addresses. If this is the case then a change should be included which causes a proper error to be surfaced in this case. Otherwise this function should be updated to to a name lookup in order to resolve an IP address.